src/rkc_utils.jl

```diff
@@ -70,7 +79,7 @@ function maxeig!(integrator, cache::OrdinaryDiffEqConstantCache)
       z = uprev + quot*tmp
     else
       # An arbitrary change on `z`
-      cache.z = shuffle(cache.z)
```

Comment:
We have used this `maxeig!` for ROCK2, ROCK4, RKC. The `cache` here is `ConstantCache` and it doesn't have any field `z`, thus I'm getting an error. I tried changing `cache.z` to `z`, then I get an error that `shuffle not defined`.

Reply:
that's for sure going to break the other algorithms @YingboMa

Reply:
Change it to `Base.shuffle`?

Reply:
I made these changes as a temporary alternative
```
nind = length(uprev)
if (nind != 1)
     ind = 1 + iter % nind
     z[ind] = uprev[ind] - (z[ind] - uprev[ind])
else
     z = -z
end
```